### DISCUSSION TITLE
Copy, template, or choose project fields from other projects

### DISCUSSION THREAD

**Discussion by szeitlin** [community/community#6022] [Projects and Issues] [answered]
I've made two projects so far with the new projects beta version, and each time I have to create all the fields manually, even though I want to use the same ones for my latest project. 

It would be great if there could be a way to choose fields from other projects, or create templates for these.

**Comment by rebeccabrandt** [comment]
Yes pleeeeeease this 👆🏼 It's so frustrating having to recreate a project and all of the fields manually. I'd love to be able to save a project as default, or at least duplicate a project and have all of the fields move over too.

**Comment by alexcnichols** [comment]
One feature that's on the public roadmap is fields that are at the issue-level that could be used within _any_ project. Although that doesn't directly hit on "project templates", it'd be interesting to see how that could be used instead or as well. 

- https://github.com/github/roadmap/issues/277

**Comment by rebeccabrandt** [comment]
@alexcnichols that actually would work very well for my workflow, and then the only field that would need to change would be the project/milestone. I like it!

Still, it's frustrating to have to recreate all of the various project views, considering our last project had around 10.

**Comment by GRobichaud** [comment]
Basically, the same copy feature as the old projects would work for me
![Screenshot from 2021-11-19 14-35-35](https://user-images.githubusercontent.com/35074602/142681466-47757858-2203-4281-a123-af24e6380338.png)

**Reply by GRobichaud** [reply]
It ended up being unecessary for me. I used the old project boards for my 2 weeks sprints, closing the current board and spawning a new one every 2 weeks. So yeah, this used to be super important to just clone a board.

After exploring the new project board, I managed to get what I want from a single "rolling" board. The first step was to "close" a sprint by removing closed tickets. It turned out it's very simple, just archiving everything in my Done column:
![image](https://user-images.githubusercontent.com/35074602/194326769-4ba14ae3-903e-4687-bfc5-a2c0e4d7af42.png)

Then, for loading fresh tickets, we just added a custom field for "Sprint", with values:
![image](https://user-images.githubusercontent.com/35074602/194327094-f150c390-631e-479e-8ec0-c8cd9aba15d6.png)

So my default "Sprint Board" is filtered on "sprint:"Current" with swimlanes on "Status". Whereas my "Planning" board is filtered on "-sprint:"Current", grouped by sprint. So I can see everything targeting the "Next" sprint and I can just switch then in one place. Again, much easier than before.
![image](https://user-images.githubusercontent.com/35074602/194328197-e9eae6bf-829a-4f8a-bb15-abef75f276fe.png)

I also like the fact I can change that straight from witin the ticket.
![image](https://user-images.githubusercontent.com/35074602/194328350-df4ddf43-ef34-4a59-ac6d-6b07eb3cd2cd.png)


So, basically, I do not need templates anymore :smile: . The remaining need maybe would be to have similar boards accross teams, but that's a much lesser concern. Leads all have their own preferences anyways...

**Reply by skipjack** [reply]
For historical/reporting purposes it seems a little dangerous to continue mutating older projects to repurpose them for new ones. I wonder if GitHub even exposes an audit log for projects through their API or UI. That's just the initial concern that comes to mind when reading your comment, but I see how this workaround would address much of the need.

**Reply by rebeccabrandt** [reply]
I totally agree with this @skipjack! Sounds scary to me. I often look back at old projects for reporting purposes.

**Comment by mbcann01** [comment]
Being able to create project templates would be really useful for me!

**Comment by mlcaraldi** [comment]
The ability to copy or duplicate a project would be really useful!

**Comment by brontevdh** [comment]
Hi :wave:

Thanks for these posts 😄, we're keen to understand more about your top use cases for project templates. Please see a quick [survey here](https://github.com/orgs/community/discussions/35253). We'd love your input.

Thanks!

**Comment by rileybroughten** [comment, marked as answer]
Following up on this that templates for Projects are now in public beta! Check out the [changelog](https://github.blog/changelog/2023-05-25-github-issues-projects-may-25th-update/#%F0%9F%8E%A8-project-templates-for-organizations) announcement for more details! ✨
